lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -722,6 +722,9 @@ def test_cache_map_traverse_cmap(
             )
             tb.cache_map(tar_dir)
 
+            if str(resolve_path).endswith("subdir15"):
+                resolve_path = tar_dir / resolve_path
+
```

Comment:
You have *three* parametrization entries that specify a non-null `resolve_path` and exactly *one* meets this condition. I'm not going to object specifically, but at the same time I don't really like such a special case here that assumes the target has to be a specific directory.

A more general mechanism might be to make the `resolve_path` a templated string, e.g., with `"{dir}subdir1/subdir15"` and `resolve_path = resolve_path.format(dir=tar_dir)`, which would allow other test cases involving an absolute path within the unpacked tarball.

Reply:
OK, I now understand why we needed this code.  I have a [suggestion](https://github.com/distributed-system-analysis/pbench/pull/3396#discussion_r1204872265) below for making it more expressive and somewhat more general, but I like Dave's suggestion here as well:  if the parameter for `resolve_path` is a string (instead of a `Path`) and if the string contains `{dir}` at the front _only_ when it is an absolute resolution, then you can construct the comparison value as Dave suggests (`expected_value = Path(resolve_path.format(dir=str(tar_dir)))`):  if the `{dir}` _isn't_ present then the `format()` will have no effect, but if it is present then the code will insert the absolute portion of the path to the unpack area...and, no `if` statement is required in the test code.